src/checkers/inference/model/InequalityConstraint.java

```diff
@@ -36,6 +41,12 @@ protected static Constraint create(Slot first, Slot second, AnnotationLocation l
                     : AlwaysFalseConstraint.create();
         }
 
+        // V == V => FALSE
+        if (first instanceof VariableSlot && second instanceof VariableSlot
```

Comment:
shouldn't need instanceof VS since everything in our system is a VS or a subclass of VS